File: 2023/java/src/test/java/org/frieder/aoc/day3/TestDay3A.java
package org.frieder.aoc.day3;

import org.frieder.aoc.day3.a.Solution3A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay3A
{
    @Test
    public void testExample() throws IOException {
        assertEquals(4361, Solution3A.getResult("../resources/3/test_a.txt"));
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(533784, Solution3A.getResult("../resources/3/input.txt"));
    }
}

File: 2023/php/test/TestDay3A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day3\a\Solution3A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay3A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(4361, Solution3A::getResult("../resources/3/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(533784, Solution3A::getResult("../resources/3/input.txt", TestLogger::createTestLogger()));
    }

}